### PR TITLE
start_method added to parallel executor

### LINKED ISSUE #882: Allowing to switch start_method in Parallel executor
I have some Pytorch code which I want to run on GPU (`net.cuda()` and whatnot)

I get the following error:
```
RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method
```
I tried to fix this by either calling
```
mp.set_start_method('spawn', force=True)
```

or 
```
torch.multiprocessing.set_start_method('spawn')
```

In the first case nothing happens, and in the second I get `RuntimeError: context has already been set`

How to set multiprocessing start method with ploomber?

### PR DESCRIPTION
**Pull request by yafimvo**
## Describe your changes
Added a new start_method argument to Parallel.py and then initiated ProcessPoolExecutor with mp_context where context is default or the given start_method

* I can't connect to GPU locally so I need to test this code on AWS VM.

## Issue ticket number and link
Closes #882 

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [ ] I have added thorough tests (when necessary).
- [ ] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

**Comment by yafimvo** [comment]
I've successfully reproduced this exception on AWS VM running the code on GPU with Cuda.
After running the new code it seems to work OK using following pipeline.yaml configuration:
```
executor:
  dotted_path: ploomber.executors.Parallel
  processes: 4
  start_method: spawn 
```

I also added a ValueType exception when `start_method` is incorrect (possible values are: `spawn, fork, forkserver`)

I'm not sure how to test this parallel behavior.

**Comment by idomic** [comment]
Nice!

> I'm not sure how to test this parallel behavior.

You'd need to run a pipeline.yaml and tasks that has independent tasks that can run in parallel.
Check out [this grid feature](https://github.com/ploomber/projects/blob/cbcf4aee763d2eac33d0d749faa3807c9b7c07d9/cookbook/grid/README.md) pipeline.

File: tests/dag/test_dag.py [hunk 1]
```diff
@@ -993,6 +988,15 @@ def test_early_stop_from_task_level_on_finish(executor, tmp_directory):
 # test early stop when registered an on_failure hook, maybe don't run hook?
 
 
+@pytest.mark.parametrize('start_method', ["invalid_start_method_value"])
```
Comment (idomic) [inline tests/dag/test_dag.py:991]: Also add a test on the other start methods and validate it's set in the context

**Review by idomic — CHANGES_REQUESTED** [review]
Looks good small comment, and missing the manual test on parallel execution

**Review by idomic — APPROVED** [review]
Nice work!
For next PRs let's try starting with some tests and then develop (TDD)

**Pull request merged**